app/lib/common/dialogs/member_info_drawer.dart

```diff
@@ -0,0 +1,264 @@
+      },
+      child: Center(
+        child: Container(
+          decoration: BoxDecoration(
+            borderRadius: BorderRadius.circular(100),
+            border: Border.all(
+              width: 2,
+              color: Theme.of(context).colorScheme.onSurface,
+            ),
+          ),
+          child: ActerAvatar(
+            mode: DisplayMode.DM,
+            avatarInfo: AvatarInfo(
+              uniqueId: memberId,
+              avatar: memberProfile.getAvatarImage(),
+              displayName: memberProfile.displayName,
+            ),
+            size: 50,
+          ),
+        ),
+      ),
+    );
+  }
+
+  Widget _buildMessageUser() {
+    return Consumer(
+      builder: (context, ref, widget) {
+        final client = ref.watch(alwaysClientProvider);
+        final dmId = client.dmWithUser(memberId).text();
+        if (dmId != null) {
+          debugPrint('DM id; $dmId');
+          return Center(
+            child: OutlinedButton.icon(
+              icon: const Icon(Atlas.chats_thin),
+              onPressed: () async {
+                context.pop();
+                context.pushNamed(
+                  Routes.chatroom.name,
+                  pathParameters: {
+                    'roomId': dmId,
+                  },
+                );
+              },
+              label: const Text('Message'),
+            ),
+          );
+        } else {
+          return Center(
+            child: OutlinedButton.icon(
+              icon: const Icon(Atlas.chats_thin),
+              onPressed: () {
+                final profile = member.getProfile();
+                ref.read(createChatSelectedUsersProvider.notifier).state = [
+                  profile,
+                ];
+                context.pop();
+                context.pushNamed(
+                  Routes.createChat.name,
+                );
+              },
+              label: const Text('Start DM'),
+            ),
+          );
+        }
+      },
+    );
+  }
+
+  Widget _buildDisplayName(BuildContext context) {
+    return GestureDetector(
+      onTap: () async {
+        await gotoFullProfile(context);
+      },
+      child: Center(
+        child: Text(memberProfile.displayName!), // FIXME: make this prettier
+      ),
+    );
+  }
+
+  Widget _buildUserName(BuildContext context) {
+    return GestureDetector(
+      onTap: () async {
+        context.pop(); // close the drawer
+        Clipboard.setData(
+          ClipboardData(
+            text: memberId,
+          ),
+        );
+        customMsgSnackbar(
+          context,
+          'Username copied to clipboard',
+        );
+      },
+      child: Row(
+        mainAxisAlignment: MainAxisAlignment.center,
+        children: [
+          Text(memberId), // FIXME: make this prettier
+          const SizedBox(width: 5),
+          const Icon(Icons.copy_outlined),
+        ],
+      ),
+    );
+  }
+
+  Future<void> gotoFullProfile(BuildContext context) async {}
```

Comment:
Can I have more information about `gotoFullProfile` ?

Reply:
Extend the drawer to full-screen should be fine if we need full page but I think we don't have that much user information to show at the moment. So we can remove `gotoFullProfile` as of now and later stage we can add it whenever required.